src/app/WriteHandler.cpp

```diff
@@ -769,6 +772,10 @@ DataModel::ActionReturnStatus WriteHandler::CheckWriteAllowed(const Access::Subj
     //       and tests and implementation
     //
     //       Open issue that needs fixing: https://github.com/project-chip/connectedhomeip/issues/33735
+
+    // First check things that are not even in metadata. These are read.only.
+    VerifyOrReturnValue(!IsSupportedGlobalAttributeNotInMetadata(aPath.mAttributeId), Status::UnsupportedWrite);
```

Comment:
Is this runtime optimization of some sort?  Because it's not clear to me why this would be needed for correctness, and it seems like extra code...  And I am not sure why we would want to runtime-optimize this rare case at the cost of slower execution for common cases and more code.

Reply:
Previous code was doing `FindMetadata` where it could return:
   - Error (immediate return value)
   - Attribute Metadata (valid attribute)
   - Cluster (assuming it would be used for global list attributes)
 
 And then the cluster would fall over in GlobalAttributeRead and return `UnsupportedWrite`.
 
 But now hoisting this up, The `AttributeFinder` is similar (has no global inside) but we need to handle globals. So I placed it here since it seems a fast check. I could have placed it in the `!attributeEntry.has_value` on line 787 instead, but that seemed less readable.
 
 Without this, Write(AttributeList) for example returns `UnsupportedAttribute` (like 787) instead of `UnsupportedWrite`

Reply:
#37345 drops the entire `cluster pointer` return from findmetadata, as it is only used for global list attributese. This and that PR are somewhat related/overlapping.

Reply:
Yes, I did notice a bunch of duplicated code between the two PRs....  Let's get that one in and then see what happens here, I guess?  Because just like in the other PR this line seems wrong for a write to a global attribute on a non-existent cluster/endpoint.

Reply:
Updated post merge for that cluster.